extensions/_modules/boto_ec2.py

```diff
@@ -1923,14 +1925,31 @@ def detach_volume(volume_id, instance_id=None, device=None, force=False,
     .. code-block:: bash
 
         salt-call boto_ec2.detach_volume vol-12345678 i-87654321
+        salt-call boto_ec2.detach_volume myvolume
 
     '''
     conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    if volume_id:
     try:
         return conn.detach_volume(volume_id, instance_id, device, force)
     except boto.exception.BotoServerError as e:
         log.error(e)
         return False
+    elif volume_name:
+        try:
+            volumes = conn.describe_volumes(Filters = [
+                                    {'Name': 'name',
+                                    'Values': [volume_name, ]}])
```

Comment:
Rather than relying on the `name` tag, we may actually want to allow for passing in a `tags` dictionary similar to https://github.com/mitodl/salt/blob/develop/salt/modules/boto_ec2.py#L580-L582